hnn_core/gui/gui.py

```diff
@@ -1118,9 +1130,45 @@ def load_drive_and_connectivity(params, log_out, drives_out,
                       layout)
 
 
-def on_upload_change(change, params, tstop, dt, log_out, drive_boxes,
-                     drive_widgets, drives_out, connectivity_out,
-                     connectivity_textfields, layout, load_type):
+def on_upload_data_change(change, data, viz_manager):
+    if len(change['owner'].value) == 0:
+        logger.info("Empty change")
+        return
+
+    key = list(change['new'].keys())[0]
+
+    data_fname = change['new'][key]['metadata']['name'].rstrip('.txt')
+    if data_fname in data['simulation_data'].keys():
+        logger.error(f"Found existing data: {data_fname}.")
+        return
+
+    ext_content = change['new'][key]['content']
+    ext_content = codecs.decode(ext_content, encoding="utf-8")
+    # just load the first column, single trial.
+    dpl_data = np.loadtxt(io.StringIO(ext_content), dtype=float)
+    if dpl_data.shape[1] > 2:
+        logger.error("New GUI does not support multi-trial data.")
+        return
+
+    data['simulation_data'][data_fname] = {'net': None, 'dpls': [
+        hnn_core.read_dipole(io.StringIO(ext_content))
+    ]}
```

Comment:
Why not just put this check in `read_dipole` so that  we can close #520 and #617?